nuitka/tools/release/Documentation.py

```diff
@@ -28,68 +28,65 @@
 from nuitka.utils.FileOperations import getFileList
 
 
+def which(prog):
```

Comment:
We already have `nuitka.utils.Execution.getExecutablePath` to which some of these changes could be considered an improvement for sure.

Reply:
I'm not sure what you mean here. Would you like me to use `nuitka.utils.Execution.getExecutablePath` here? Or change `nuitka.utils.Execution.getExecutablePath` ?

Reply:
Use it, merging the enhancements, e.g. to use `shutil.which` where possible. But clearly not to duplicate the kind of code.

Reply:
You mean something like this?

```
def getExecutablePath(filename):
    """ Find an execute in PATH environment.
    """

    # Append ".exe" suffix  on Windows if not already present.
    if getOS() == "Windows" and not filename.lower().endswith(".exe"):
        filename += ".exe"

    if sys.version_info >= (3, 3):
        import shutil
        return shutil.which(filename)
    else:
        # Search in PATH environment.
        search_path = os.environ.get("PATH", "")

        # Now check in each path element, much like the shell will.
        path_elements = search_path.split(os.pathsep)

        for path_element in path_elements:
            path_element = path_element.strip('"')

            full = os.path.join(path_element, filename)

            if os.path.exists(full):
                return full
```